### PR TITLE
Fix kafka broker extension

### PR DESCRIPTION
**Pull request by ozangunalp**
- Fix for junit extension for Kafka broker, which creates a single broker for all the test suite
- KafkaCompanion producer and consumer creation accept untyped Serde classes for configuring generic Serde ex. avro

### CODE REVIEW

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: smallrye-reactive-messaging-kafka-test-companion/src/main/java/io/smallrye/reactive/messaging/kafka/companion/ConsumerBuilder.java [hunk 1]
```diff
@@ -149,8 +149,9 @@ public ConsumerBuilder(Map<String, Object> props, Duration kafkaApiTimeout,
      * @param keyDeserType the deserializer class for record keys
      * @param valueDeserType the deserializer class for record values
      */
+    @SuppressWarnings("rawtypes")
     public ConsumerBuilder(Map<String, Object> props, Duration kafkaApiTimeout,
-            Class<? extends Deserializer<K>> keyDeserType, Class<? extends Deserializer<V>> valueDeserType) {
+            Class<? extends Deserializer> keyDeserType, Class<? extends Deserializer> valueDeserType) {
```
[the remark this answers is not in the document]
Comment (ozangunalp) [reply smallrye-reactive-messaging-kafka-test-companion/src/main/java/io/smallrye/reactive/messaging/kafka/companion/ConsumerBuilder.java:154]: I was aiming for type erasure but yes `Class<? extends Deserializer<?>>` would work perfectly.

File: smallrye-reactive-messaging-kafka-test-companion/src/main/java/io/smallrye/reactive/messaging/kafka/companion/test/KafkaBrokerExtension.java [hunk 2]
```diff
@@ -30,16 +29,16 @@
...
 
-    private static boolean started = false;
-    protected static StrimziKafkaContainer kafka;
+    protected StrimziKafkaContainer kafka;
 
     @Override
     public void beforeAll(ExtensionContext context) {
-        if (!started) {
+        ExtensionContext.Store globalStore = context.getRoot().getStore(GLOBAL);
+        KafkaBrokerExtension extension = (KafkaBrokerExtension) globalStore.get(KafkaBrokerExtension.class);
```
[the remark this answers is not in the document]
Comment (ozangunalp) [reply smallrye-reactive-messaging-kafka-test-companion/src/main/java/io/smallrye/reactive/messaging/kafka/companion/test/KafkaBrokerExtension.java:35]: It is used like any other key on the global store here. The fix is to stop using static fields on the `KafkaBrokerExtension` class for having a single broker instance for a test suite.

It was difficult to reproduce local but I assume the original issue was that when a surefire fork is reused it _can_ execute a different test suite that uses the same extension, with its static fields.

**Comment by ozangunalp** [comment]
@cescoffier I am starting to think that it is still too early to use kraft mode in tests

**Comment by ozangunalp** [comment]
This along with https://github.com/strimzi/test-container/pull/31 will help to stabilise the tests.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**